src/FSharpVSPowerTools.Core/AssemblyContentProvider.fs

```diff
@@ -184,29 +184,33 @@ module AssemblyContentProvider =
             |> Seq.map (traverseEntity contentType Parent.Empty)
             |> Seq.concat
             |> Seq.distinct
-            |> Seq.toList
 
     let private getAssemblySignaturesContent contentType (assemblies: FSharpAssembly list) = 
         assemblies 
-        |> List.map (fun asm -> getAssemblySignatureContent contentType asm.Contents)
-        |> List.concat 
+        |> Seq.map (fun asm -> getAssemblySignatureContent contentType asm.Contents)
+        |> Seq.concat
+        |> Seq.toList
+
+    let private entityCache = Dictionary<AssemblyPath, DateTime * AssemblyContentType * RawEntity list>()
 
-    let private entityCache = ConcurrentDictionary<AssemblyPath, DateTime * AssemblyContentType * RawEntity list>()
+    let (|DynamicAssembly|_|) (fileName: string) = if fileName.StartsWith "tmp" then Some () else None
```

Comment:
Is there a more reliable way to recognize dynamic assemblies? Does `IsProviderGenerated` work https://github.com/fsharp/FSharp.Compiler.Service/blob/9a3fc667180ab0a04ee68d341804d40b9d34a008/src/fsharp/vs/Symbols.fsi#L101?